Add migration steps for `ConfigLoader` to `OmegaConfigLoader` (#2887)
kedro-org/kedro: docs/source/configuration/migration.md

## Migration guide for config loaders

> The `ConfigLoader` and `TemplatedConfigLoader` classes are now deprecated and scheduled for removal in Kedro `0.19.0`. To ensure smooth transitions for users, we strongly recommend adopting the [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader) promptly.

Reviewer: ```suggestion
The `ConfigLoader` and `TemplatedConfigLoader` classes have been deprecated since Kedro `0.18.x` and they scheduled for removal in Kedro `0.19.0`. To ensure a smooth transition, we strongly recommend you adopt the [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader) as soon as possible.
```

Reviewer: Maybe that is 0.18.5?

Author: Done, it's 0.18.12.

> This migration guide outlines the primary distinctions between the old loaders and the `OmegaConfigLoader`, providing step-by-step instructions on updating your code base to utilise the new class effectively.

Reviewer: ```suggestion
This migration guide outlines the primary distinctions between the old loaders and the `OmegaConfigLoader`, and provides step-by-step instructions to update your code to use the new class.
```

### 1. Install the Required Library
The [`OmegaConfigLoader`](advanced_configuration.md#omegaconfigloader) was introduced in Kedro `0.18.5` and is based on [OmegaConf](https://omegaconf.readthedocs.io/). In order to use it you need to ensure you have both a version of Kedro of `0.18.5` or above and `omegaconf` installed.
You can install both using `pip`:

```bash
pip install kedro==0.18.5
```
This would be the minimum required Kedro version which includes `omegaconf` as dependency.
Or you can run:
```bash
pip install -U kedro
```

This command installs the most recent version of Kedro which also includes `omegaconf` as dependency.

### 2. Import Statements
Replace the import statement for `ConfigLoader` with the one for `OmegaConfigLoader`:

```python
# Before:
from kedro.config import ConfigLoader

# After:
from kedro.config import OmegaConfigLoader
```

### 3. File Format Support
`OmegaConfigLoader` supports only `yaml` and `json` file formats. Make sure that all your configuration files are in one of these formats. If you were using other formats with `ConfigLoader`, convert them to `yaml` or `json`.

### 4. Load Configuration

> The method to load the configuration using `OmegaConfigLoader` is slightly updated. The `ConfigLoader` allowed users to access configuration through the `.get()` method, which required patterns as argument.

Reviewer: ```suggestion
The method to load the configuration using `OmegaConfigLoader` differs slightly from that used by `ConfigLoader`, which allowed users to access configuration through the `.get()` method and required patterns as argument.
```

Reviewer: Not sure about how to edit "patterns as argument" but that doesn't seem quite right??

Author: The `.get()` method took patterns e.g. "catalog*", or "**/catalog.yml" as argument input. But I get that's a bit vague... 🤔

> The `OmegaConfigLoader` requires you to fetch configuration through a configuration key that points to [configuration patterns specified in the loader class](configuration_basics.md#configuration-patterns) or [provided in the `CONFIG_LOADER_ARGS`](advanced_configuration.md#how-to-change-which-configuration-files-are-loaded) in `settings.py`.

Reviewer: ```suggestion
When you migrate to use `OmegaConfigLoader`it  requires you to fetch configuration through a configuration key that points to [configuration patterns specified in the loader class](configuration_basics.md#configuration-patterns) or [provided in the `CONFIG_LOADER_ARGS`](advanced_configuration.md#how-to-change-which-configuration-files-are-loaded) in `settings.py`.
```
